File: mobile/translator_app/lib/models/channel.dart
class Channel {
  Channel({
    required this.id,
    required this.eventId,
    required this.name,
    required this.languageCode,
  });

  final String id;
  final String eventId;
  final String name;
  final String languageCode;

  factory Channel.fromJson(Map<String, dynamic> json) {
    return Channel(
      id: json['id'] as String,
      eventId: json['eventId'] as String,
      name: json['name'] as String? ?? '',
      languageCode: json['languageCode'] as String? ?? '',
    );
  }
}

File: backend/Backend.Models/Entities/Channel.cs
namespace Backend.Models.Entities;

public sealed class Channel
{
    public Guid Id { get; set; }
    public Guid EventId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string LanguageCode { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }

    public Event Event { get; set; } = null!;
    public ICollection<Session> Sessions { get; set; } = new List<Session>();
}
